### DISCUSSION TITLE
mapping cell and point data deletion

### DISCUSSION THREAD

**Discussion by 19kdc3** [marcomusy/vedo#819] [Q&A] [answered]
Hi Marco,

When I use either the `map_cells_to_points` or `map_points_to_cells` functions and specify the array, I notice that it deletes all other data from the original location. Is there a way to map data without deleting other data sets?

**Example**
pointdata: ['A'], ['B'], ['C']
celldata:

map_points_to_cells(['A'])

pointdata:
celldata: ['A']

**Expected:**
pointdata: ['A'], ['B'], ['C']
celldata:

map_points_to_cells(['A'])

pointdata: ['B'], ['C']
celldata: ['A']

I'm using vedo version: 2023.4.3.dev18

Thanks!

**Comment by marcomusy** [comment, marked as answer]
Hi @19kdc3 try:

```
pip install -U git+https://github.com/marcomusy/vedo.git
```

then use `move=False`, e.g.:

```python
from vedo import *

mesh1 = Mesh(dataurl+'icosahedron.vtk').linewidth(0.1).flat()

mesh1.pointdata["Yscalars_p"] = mesh1.points()[:, 1]
mesh1.pointdata["Zscalars_p"] = mesh1.points()[:, 2]
mesh1.celldata[ "Zscalars_c"] = mesh1.cell_centers()[:, 2]

print("-------- Before:")
mesh2 = mesh1.clone(deep=False)
print("pointdata", mesh2.pointdata.keys())
print("celldata",mesh2.celldata.keys())

print("-------- After:")
mesh2.map_points_to_cells(['Zscalars_p'], move=False)
print("pointdata",mesh2.pointdata.keys())
print("celldata",mesh2.celldata.keys())
```

Output:
```
-------- Before:
pointdata ['Normals', 'Yscalars_p', 'Zscalars_p']
celldata ['Normals', 'Zscalars_c']
-------- After:
pointdata ['Normals', 'Yscalars_p', 'Zscalars_p']
celldata ['Normals', 'Zscalars_c', 'Zscalars_p']
```

**Reply by 19kdc3** [reply]
Awesome! Thanks for the quick fix Marco!
